include/nbl/builtin/hlsl/subgroup/fft.hlsl

```diff
@@ -32,6 +32,7 @@ template<typename Scalar, uint32_t SubgroupSize, bool inverse>
 void FFT(NBL_REF_ARG(complex_t<Scalar>) lo, NBL_REF_ARG(complex_t<Scalar>) hi) 
```

Comment:
you shouldn't template on the subgroup size, just use `glsl::gl_SubgroupSize()` for your loops

Reply:
you need to change the template signature to
```cpp
template<bool Inverse, typename Scalar>
```
the reasons is you need to put non-deductible template parameters first (otherwise I need to type out the `Scalar` type. Also the subgroup size as above, can't be a constexpr.

Furthermore I'd experiment with rolling up the FFT and FFT_loop into a single struct with an `operator()` to keep convention with the subgroup arithmetic operations